src/user_input_compiler.rs

```diff
@@ -219,9 +217,7 @@ fn match_duration(s :&mut Chars) -> Result<Duration,Input> {
 fn match_real(s: &mut Chars) -> Result<f64, Input> {
   let st :String = try!(collect_real(s));
   match st.parse::<f64>() {
-    Ok(val) => {
-      Ok(val)
-    },
+    Ok(val)  => Ok(val),
     Err(msg) => Err( InvalidSyntax(format!("Invalid Real: {:?}", msg)) )
   }
```

Comment:
In cases like this, where you only do processing in one of the two cases, it's almost always possible to use combinator methods:

```rust
st.parse::<f64>().map_err(|e| InvalidSyntax(format!("Invalid real: {:?}", e)))
```